lib/src/signed_video_h26x_auth.c

```diff
@@ -744,30 +783,43 @@ maybe_validate_gop(signed_video_t *self, h26x_nalu_t *nalu)
 
   svi_rc status = SVI_UNKNOWN;
   SVI_TRY()
-    SVI_THROW(prepare_for_validation(self));
+    // |first_validation_check| keeps track of first validation when looping through pending gops
+    bool first_validation_check = true;
+    // Loop through possible pending gops and validate them
+    for (int i = 0; i < nalu_list->gop_idx; i++) {
```

Comment:
With a fixed list we can blow the limit if the user sets a very long recurrence, which you have a check for on line 743. Also, if we lose the SEIs with the public_key number of pending GOPs will/can grow past the limit.

We should consider using a linked list instead.
Unfortunately the auth-side does not know what recurrence was used in camera. Could we benefit from transmitting the recurrence interval in the general tag?